dali/sasha/saarch.cpp

```diff
@@ -85,295 +85,444 @@ static void mkDateCompare(bool dfu,const char *dt,StringBuffer &out,char fill)
-            StringBuffer val;
-            ForEach(*iter) {
-                IPropertyTree &pt=iter->query();
-                const char *wuid = pt.queryName();
-                if (index>start+num)
-                    break;
-        //      PROGLOG("match before=%s after=%s wuid=%s",before.str(),after.str(),wuid);
-                if ((!mask||!*mask||!isWild||WildMatch(wuid,mask,true)) &&
-                    ((before.length()==0)||(stricmp(wuid,before)<0)) &&
-                    ((after.length()==0)||(stricmp(wuid,after)>=0))) {
-        //          PROGLOG("matched before=%s after=%s wuid=%s",before.str(),after.str(),wuid);
-                    bool hasowner = owner&&*owner;
-                    bool hascluster = cluster&&*cluster;
-                    bool hasstate = state&&*state;
-                    bool hasjobname = jobname&&*jobname;
-                    bool hasoutput = outputformat&&*outputformat;
-                    bool inrange = (index>=start)&&(index<start+num);
-                    if (hasowner||hasstate||hascluster||hasjobname||(hasoutput&&inrange)) {
-                        try {
-                            if (hasowner&&(!pt.getProp("@submitID",val.clear())||!WildMatch(val.str(),owner,true)))
-                                continue;
-                            if (hasstate&&(!pt.getProp("@state",val.clear())||!WildMatch(val.str(),state,true)))
-                                continue;
-                            if (hascluster&&(!pt.getProp("@clusterName",val.clear())||!WildMatch(val.str(),cluster,true)))
-                                continue;
-                            if (hasjobname&&(!pt.getProp("@jobName",val.clear())||!WildMatch(val.str(),jobname,true)))
+            setSDSConnection(conn, mask);
+            return isWild ? conn->queryRoot()->hasProp(wuid) : conn != nullptr;
+        }
+        bool checkFilters(IPropertyTree *t, bool onlineWU)
+        {
+            //The following code is a clone of the pre-existing code.
+            const char *owner = cmd->queryOwner();
+            const char *cluster = cmd->queryCluster();
+            const char *jobName = cmd->queryJobName();
+            const char *state = cmd->queryState();
+
+            StringBuffer path, val;
+            if (!isEmptyString(owner) && (!t->getProp("@submitID", val.clear()) || !WildMatch(val.str(), owner, true)))
+                return false;
+            if (!isEmptyString(state) && (!t->getProp((!onlineWU && cmd->getDFU()) ? "Progress/@state" : "@state", val.clear()) || !WildMatch(val.str(), state, true)))
+                return false;
+            if (!isEmptyString(cluster) && (!t->getProp("@clusterName", val.clear()) || !WildMatch(val.str(), cluster, true)))
+                return false;
+            if (!isEmptyString(jobName) && (!t->getProp("@jobName", val.clear()) || !WildMatch(val.str(), jobName, true)))
+                return false;
+
+            if (onlineWU)
+                return true;
```

Comment:
I'm a bit confused by this, the workunit search dialog blanks out the ECL and logicalfile fields, if "Archive Only" is ticked.
Implying that you can use these fields for online workunits, but not archived workunits.

However, checkFilters with onlineWU=true, is used getOnlineWUs() and onlineWU=false for the archived version, so it's applying these filters to archived workunits only.

Is the search dialog wrong?
And can you only search by these fields in archived workunits?

Reply:
I am using the onlineWU flag to simulate the old code (saarch.cpp, line 179-215 and 310 - 326). This part of the code is not used by ECLwatch's online WU search.